lib/tasks/integration.rake

```diff
@@ -1,144 +1,176 @@
+# frozen_string_literal: true
+
 require 'color'
+require 'benchmark'
 
-desc 'Run continuous integration'
-task :integrate, :log do |_, args|
-  if ENV['CI']
-    ENV['COVERAGE'] = '1'
-    ENV['PERCY_ENABLE'] = '0' # percy will be enabled just for one task
-  end
 
-  abort 'failed to run integrate:prepare' unless system('rake integrate:prepare --trace')
+def print_banner_around
```

Comment:
Something to bear in mind: defining methods like this in Rake defines them in a global `main` module, so they are shared between all tasks. It is quite unfortunate, but that's what we have.

https://kevinjalbert.com/defined_methods-in-rake-tasks-you-re-gonna-have-a-bad-time/ sumarizes it well. 

Another option not described in that post is using lambdas in local variables.

Reply:
great read, didn't know, thanks!! ;)